MeesGame/Gameplay/UIObjects/UIObject.cs

```diff
@@ -40,11 +158,13 @@ public bool Clicked
             get { return clicked; }
         }
 
-        public bool Invalidate
+        public virtual bool Invalidate
```

Comment:
Please make this a function Invalidate(). If it is not a function it should not be a verb, it should be Invalidated.
Also, add documentation to this property.